test/livebook/live_markdown/markdown_helpers_test.exs

```diff
@@ -108,6 +190,42 @@ defmodule Livebook.LiveMarkdown.MarkdownHelpersTest do
       assert markdown == reformat(markdown)
     end
 
+    test "code block with fences inside it" do
+      markdown = """
+      ~~~~elixir
+      before
+
+      ```
+      _inside_
+      ```
+
+      after
+      ~~~~\
+      """
+
+      assert markdown == reformat(markdown)
```

Comment:
Another "unstable under repeat formatting case": the first pass of formatting alters the meaning of the fences by transforming the outside ones to ` ``` `, and the second one then treats the inside content as text instead of fenced code.

Reply:
Good catch, that's similar to #438, we should determine suitable delimiter based on the content.